docs/source/examples/ddd_tutorial.myst

```diff
@@ -0,0 +1,285 @@
+
+```{code-cell} ipython3
+# Assume chain-like connectivity
+computer = nx.Graph()
+computer.add_edges_from([(0, 1), (1, 2)])
+nx.draw(computer, with_labels=True)
+
+# Add reversed edges to computer graph.
+# This is important to represent CNOT gates with target and control reversed.
+computer = nx.to_directed(computer)
+```
+
+## Define the circuit
+
+We use mirror circuits to benchmark the performance of the device.
+Mirror circuits, introduced in
+*Proctor et al. arXiv (2021)* {cite}`Proctor_2021_NatPhys` ([arXiv:2008.11294](https://arxiv.org/abs/2008.11294)),
+are designed such that only one bitstring should be sampled. 
+When run on a device, any other measured bitstrings are due to noise.
+The frequency of the correct bitstring is our target metric.
+
+> Note: Mirror circuits build on Loschmidt echo circuits - i.e., circuits of the form $U U^\dagger$ for some unitary $U$.
+Loschmidt echo circuits are good benchmarks but have shortcomings - e.g., they are unable to detect coherent errors.
+Mirror circuits add new features to account for these shortcomings. 
+For more background, see [arXiv:2008.11294](https://arxiv.org/abs/2008.11294).
+
+To define a mirror circuit, we need the device graph. 
+We will use a subgraph of the device, and our first step is picking a subgraph with good qubits.
+
+
+### Generate mirror circuit
+
+Now that we have the device (sub)graph, we can generate a mirror circuit and the bitstring it should sample as follows.
+
+```{code-cell} ipython3
+def get_circuit(depth: int, seed: int) -> Tuple[cirq.Circuit, str]:
+    circuit, correct_bitstring = benchmarks.generate_mirror_circuit(
+        nlayers=depth,
+        two_qubit_gate_prob=1.0,
+        connectivity_graph=computer,
+        two_qubit_gate_name="CNOT",
+        seed=seed,
+        return_type="cirq",
+    )
+    return circuit, correct_bitstring
+```
+
+### Helper methods
+
+It will be useful later to us to have a number of utility functions for counting CNOT gates and operations.
+
+```{code-cell} ipython3
+def get_num_cnot_count(circuit: cirq.Circuit) -> int:
+    """Determine number of cnot gates in a given `Circuit` object."""
+    # Count CNOT gates for `cirq`-type circuit objects:
+    num_cnots: int = 0
+    for instruction in circuit.all_operations():
+        if isinstance(instruction, cirq.CXPowGate):
+            num_cnots += 1
+    return num_cnots
+
+
+def get_oneq_count(circuit: cirq.Circuit) -> int:
+    return len(list(circuit.all_operations())) - get_num_cnot_count(circuit)
+```
+
+## Define the executor
+
+Now that we have a circuit, we define the `execute` function which inputs a circuit and returns an expectation value - here, the
+frequency of sampling the correct bitstring.
+
+```{code-cell} ipython3
+def execute(
+    circuits: Union[cirq.Circuit, List[cirq.Circuit]],
+    shots: int,
+    correct_bitstring: str,
+    is_noisy: bool = True,
+) -> List[float]:
```

Comment:
I might be misunderstanding something here, but I can't find anywhere where `execute` is called with a list of circuits. If so, we might be able to simplify this function.

Reply:
Good catch, removed.